lib/ruby_lsp/standard/wraps_built_in_lsp_standardizer.rb

```diff
@@ -0,0 +1,83 @@
+module RubyLsp
+  module Standard
+    class WrapsBuiltinLspStandardizer
+      include RubyLsp::Requests::Support::Formatter
+      def initialize
+        init!
+      end
+
+      def init!
+        @config = ::Standard::BuildsConfig.new.call(ARGV)
+        @standardizer = ::Standard::Lsp::Standardizer.new(
+          @config,
+          ::Standard::Lsp::Logger.new
+        )
+        @cop_registry = RuboCop::Cop::Registry.global.to_h
+      end
+
+      def run_formatting(uri, document)
+        @standardizer.format(uri_to_path(uri), document.source)
+      end
+
+      def run_diagnostic(uri, document)
+        offenses = @standardizer.offenses(uri_to_path(uri), document.source)
+
+        offenses.map { |o|
+          cop_name = o[:cop_name]
+
+          msg = o[:message].delete_prefix(cop_name)
+          loc = o[:location]
+
+          severity = case o[:severity]
+          when "error", "fatal"
+            RubyLsp::Constant::DiagnosticSeverity::ERROR
+          when "warning"
+            RubyLsp::Constant::DiagnosticSeverity::WARNING
+          when "convention"
+            RubyLsp::Constant::DiagnosticSeverity::INFORMATION
+          when "refactor", "info"
+            RubyLsp::Constant::DiagnosticSeverity::HINT
+          else # the above cases fully cover what RuboCop sends at this time
+            logger.puts "Unknown severity: #{severity.inspect}"
+            RubyLsp::Constant::DiagnosticSeverity::HINT
+          end
+
+          RubyLsp::Interface::Diagnostic.new(
+            code: cop_name,
+            code_description: code_description(cop_name),
+            message: msg,
+            source: "Standard Ruby",
+            severity: severity,
+            range: RubyLsp::Interface::Range.new(
+              start: RubyLsp::Interface::Position.new(line: loc[:start_line] - 1, character: loc[:start_column] - 1),
+              end: RubyLsp::Interface::Position.new(line: loc[:last_line] - 1, character: loc[:last_column] - 1)
+            )
+            # TODO: do I need something like this?
```

Comment:
If you want to support quickfixes through code actions, then yes you'll need to provide those as part of data.

But you can always start without it and add later.

Reply:
Thanks for explaining. Right now I only have access to a JSON representation of the offense parsed from RuboCop over stdout, so getting the corrector will (I assume?) require us to invoke RuboCop directly in our LSP code.